snuba/datasets/transactions_processor.py

```diff
@@ -206,8 +207,8 @@ def _process_breakdown(
     def _process_contexts_and_user(
         self, processed: MutableMapping[str, Any], event_dict: EventDict,
     ) -> None:
-        contexts: MutableMapping[str, Any] = _as_dict_safe(
-            event_dict["data"].get("contexts", None)
+        contexts: MutableMapping[str, Any] = copy.deepcopy(
+            _as_dict_safe(event_dict["data"].get("contexts", None))
         )
```

Comment:
Why is this working?
This pop happens before the call to _process_contexts_and_user
https://github.com/getsentry/snuba/blob/master/snuba%2Fdatasets%2Ftransactions_processor.py#L334-L337

Reply:
Good catch. You are right that the message was missing those fields.

I have refactored the code so that all modifications of contexts happen just before we fill in the contexts key and value columns. So there should not be any more blindspots.

Also verified that the errors processor does not perform and `del` or `pop` operations.